### PR TITLE
Minor revamp

### PR DESCRIPTION
**Pull request by CommandString**

### CODE REVIEW

File: examples/HttpCodesApi/GetCode.php [hunk 1]
```diff
@@ -18,25 +18,25 @@ public static function handle(
...
+        $html = ob_get_clean();
+
+        return new HtmlResponse($html);
     }
 
-    public static function getCodes()
+    public static function getCodes(): array
     {
-        return json_decode(file_get_contents(__DIR__ . "/../../tools/HttpExceptions/HttpCodes.json"));
+        return json_decode(file_get_contents(__DIR__ . "/../../tools/HttpExceptions/HttpCodes.json"), true);
```
Comment (Exanlv) [inline examples/HttpCodesApi/GetCode.php:40]: Still using json instead of php to store the data

File: src/Router.php [hunk 2]
```diff
@@ -359,20 +359,20 @@ public static function run(EmitterInterface $emitter = null): void
             }
 
             if ($resolved === null) {
-                if (!self::$emitHttpExceptions) {
+                if (!is_subclass_of($e, HttpException::class)) {
```
Comment (Exanlv) [inline src/Router.php:362]: This is jank, just catch HttpExceptions seperately

**Review by Exanlv — CHANGES_REQUESTED** [review]

File: src/Router.php (hunk 2 above)
Comment (CommandString) [reply src/Router.php:362]: You mean just have [Catchers](https://github.com/tnapf/Router#catching)?

File: examples/HttpCodesApi/GetCode.php (hunk 1 above)
Comment (CommandString) [reply examples/HttpCodesApi/GetCode.php:40]: Well this is just an example so I wouldn't worry too much about it 😅

File: src/Router.php (hunk 2 above)
Comment (Exanlv) [reply src/Router.php:362]:
```php
try {
    // ...
} catch (HttpException $e) {
    // ...
} catch (Exception $e) {
    // ...
}
```

File: examples/HttpCodesApi/GetCode.php (hunk 1 above)
Comment (Exanlv) [reply examples/HttpCodesApi/GetCode.php:40]: youre also using this json in other places, no?

Comment (CommandString) [reply examples/HttpCodesApi/GetCode.php:40]: Just for generator stuff nothing else

File: src/Router.php (hunk 2 above)
Comment (CommandString) [reply src/Router.php:362]: YOU CAN DO THAT!?!?

**Pull request merged**